test/download.bats

```diff
@@ -34,14 +42,15 @@ no_properties_and_fetch ()               { assert_no_properties && fetch_launche
 fetch_launcher () {
   local version="$1" && shift
   stub_curl
+  stub_curl
```

Comment:
@dwijnand Something in this test framework is strange. I can't figure it out... It seems I have to stub curl twice, once for each potential invocation (curl JAR + curl MD5).
But I cannot keep the unstub instances. Those will fail if I keep them around.

Similarly, the unstub in the `teardown` throws an error if you get rid of the `rm` line.

Reply:
Yeah, it's confusing as hell. 😄

What do you mean by "I cannot keep the unstub instances"?

I can have a look at this detail (if you want) after you get back to me.

Reply:
I removed all the `unstub curl` stuff because it will do an `exit 1`, which fails the test.
I could not figure out how to make "unstubbing" work.

Reply:
OK, I'll have a look.

Reply:
Hey, @NeQuissimus, sorry for the delay.

So the `stub` machinery is very strict.  Once you use a stub, it's used, which is why you need two stubs for each curl usage _and_ `unstub` is important as it verifies that you used everything that you stubbed.

I've got a local version working where the fundamental diff is this:

```diff
-stub_curl() { stub curl "$curl_opts"; }
+stub_curl() {
+  stub curl "$curl_opts";
+  case "$1" in
+    0.*) ;; # sbt <1 has no MD5 files to curl and verify
+      *) stub curl "$curl_opts" && stub md5sum 'true' ;;
+  esac
+}
```

Let me know how that works, I'll circle back to this in a few days.